binary-array-ld-cli/src/main/kotlin/net/bald/BinaryArrayConvertCli.kt

```diff
@@ -1,14 +1,65 @@
 package net.bald
 
+import net.bald.model.ModelBinaryArrayConverter
+import net.bald.netcdf.NetCdfBinaryArray
+import org.apache.commons.cli.DefaultParser
+import org.apache.commons.cli.HelpFormatter
+import org.apache.commons.cli.Options
+import java.io.File
+import java.io.OutputStream
+import kotlin.system.exitProcess
+
 /**
  * Command Line Interface for converting NetCDF metadata to Linked Data graphs.
  */
 class BinaryArrayConvertCli {
+    private val opts = Options().apply {
+        this.addOption("u", "uri", true, "The URI which identifies the dataset.")
+        this.addOption("h", "help", false, "Show help.")
+    }
+
     fun run(vararg args: String) {
-        TODO()
+        val cmdOpts = options(opts, *args)
+        if (cmdOpts.help) {
+            help()
+        } else {
+            try {
+                doRun(cmdOpts)
+            } catch (e: Exception) {
+                help()
+                throw e
+            }
+        }
+    }
+
+    private fun doRun(opts: CommandLineOptions) {
+        val inputLoc = opts.inputLoc ?: throw IllegalArgumentException("First argument is required: NetCDF file to convert.")
+        val ba = NetCdfBinaryArray.create(inputLoc, opts.uri)
+        val model = ba.use(ModelBinaryArrayConverter::convert)
+
+        modelOutput(opts.outputLoc).use { output ->
```

Comment:
Just to check my understanding of use. Is this equivalent to model.write(modelOutput(opts.outputLoc), "ttl"), apart from  the output of  modelOutput(opts.outputLoc) is closed afterwards?
Please let me know if I've completely misunderstood!

Reply:
That's correct - `use` effectively replaces "try with resources" in Java.